pvl_singlediode.m

```diff
@@ -338,13 +338,19 @@
 p = (A+B)./2;
 err = g(p,Iph,Io,a,Rs,Rsh);  % value of dP/dV at initial guess p
 
+iterCount = 0;
 while max(abs(B-A))>1e-6   % set precision of estimate of Imp to 1e-6 (A)
     gA=g(A,Iph,Io,a,Rs,Rsh); % value of dP/dV at left endpoint
     u=(gA.*err)<0;
     B(u)=p(u);
     A(~u)=p(~u);
     p=(A+B)/2;
     err = g(p,Iph,Io,a,Rs,Rsh);
+    
+    iterCount = iterCount + 1;
+    if iterCount > 1000
```

Comment:
I don't object to adding `iterCount` as insurance against some kind of erroneous behavior in the helper function g(), but perhaps reduce the limit to 50.  In this bisection, convergence is achieved at an interval of width < 1e-6, from a starting interval between 0 and Iph.  A reasonably large value of Iph is 10.  In this case convergence is reached after ~20 iterations, assuming the code functions as intended.